### DISCUSSION TITLE
[Uno.Wasm.Bootstrap] RuntimeError on P/Invoke into SDL

### DISCUSSION THREAD

**Discussion by Beyley** [unoplatform/uno#10293] [Q&A] [answered]
In my long journey into proper P/Invoke on WASM support in Silk.NET, ive gotten the code compiling and launched, with the generator using P/Invoke internally, specifically
```cs
            [System.Runtime.InteropServices.DllImport("SDL", EntryPoint = "SDL_Init", CallingConvention = (System.Runtime.InteropServices.CallingConvention)2)]
            private static extern int I_SDL_Init(uint p0);
``` 
in this case

But upon attempting to call said function (more specifically a public wrapper over said function) i get a `Uncaught (in promise) RuntimeError: index out of bounds` error
full stack:
https://gist.github.com/Beyley/7274402d5e8af93e77ef933f8311511d

Test project that calls it: https://github.com/Beyley/UnoPInvoke
And here are the libs that are referenced in the csproj (just modify it to reference the DLLs directly if you need to build locally) [Silk.NET..tar.gz](https://github.com/unoplatform/uno/files/9896898/Silk.NET.tar.gz)

Heres the full generated code
https://gist.github.com/Beyley/94cf60a52ba39244b3eb5d615398e42d

The public wrapper of said function is created by getting a function pointer to the external function
`"SDL_Init" => (System.IntPtr)(delegate* managed<uint, int> )(&I_SDL_Init),`
Then calling that 
```cs
        public partial int Init(uint flags)
        {
            unsafe
            {
                int n1;
                SilkMarshal.Log(); // BEGIN RESOLVE 1 | READCOUNT: 1 | ALLOW_INLINE: False | FORCE_INLINE: False | CACHED: False
                // BEGIN RESOLVE 0 | READCOUNT: 1 | ALLOW_INLINE: True | FORCE_INLINE: False | CACHED: False
                // RESOLVED 0 INLINED
                // END RESOLVE 0
                // RESOLVED 1 TO VARIABLE n1
                n1 = ((delegate* unmanaged[Cdecl]<uint, int> )(CurrentVTable as GeneratedVTable).SDL_Init)((/* VAR 0 */
                flags)); // END RESOLVE 1
                // ALREADY RESOLVED 1
                return /* VAR 1 */
                n1;
            }
        }
```

**Comment by Beyley** [comment, marked as answer]
Seems to be caused by not haivng a UnmangedCallersOnly attribute (which is not on netstandard 2.0), so this may just not work on net standard 2.0

**Reply by jeromelaban** [reply]
You can write your own, the pinvoke generator does not validate the origin. As long as it as the same qualified name (namespace.class), you can create an internal type and use it.
